dashboard/app/models/script.rb

```diff
@@ -607,12 +609,14 @@ def csf_finish_url
     end
   end
 
-  def summarize
+  def summarize(user = nil)
```

Comment:
I think should keep `Script#summarize` anonymous and stuff user-specific data into [`summarize_user_progress`](https://github.com/code-dot-org/code-dot-org/blob/ddb66dda16dbbec9a7c522c2c3465af59cf7a0c3/dashboard/app/helpers/users_helper.rb#L8).